### PR TITLE
Only allow one module per file

### PR DESCRIPTION
**Pull request by garyb**
@paf31 I have a bit of time after all :wink:

This is pretty straightforward, mostly updates to the docs tests to make them parse.

### CODE REVIEW

**Comment by hdgarrood** [comment]
:+1: lgtm

File: src/Language/PureScript/Parser/Declarations.hs [hunk 1]
```diff
@@ -258,29 +257,31 @@ parseModule = do
...
-parseModulesFromFiles toFilePath input = do
-  modules <- flip parU id $ map wrapError $ inParallel $ flip map input $ \(k, content) -> do
+parseModulesFromFiles
+  :: forall m k
+   . MonadError MultipleErrors m
+  => (k -> FilePath)
+  -> [(k, String)]
+  -> m [(k, Module)]
+parseModulesFromFiles toFilePath input =
+  flip parU id $ map wrapError $ inParallel $ flip map input $ \(k, content) -> do
```
Comment (paf31) [inline src/Language/PureScript/Parser/Declarations.hs]: Non-blocking: this can be simplified to `flip parU wrapError . inParallel . flip map input`.

**Comment by paf31** [comment]
👍 Thanks! I started this last night but stopped when I realized I had to change the examples 😄

**Comment by paf31** [comment]
👍 Looks good to merge.

**Pull request merged**